File: history/src/main/java/com/nurupo/movie/history/entity/ResponseJSON.java
package com.nurupo.movie.history.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class ResponseJSON {
    @JsonProperty("status")
    private int status;

    @JsonProperty("obj")
    private Object obj;

    @JsonProperty("msg")
    private String msg;

    public ResponseJSON() {
        this(0);
    }

    public ResponseJSON(int status) {
        this(status, null);
    }

    public ResponseJSON(int status, Object obj) {
        this(status, obj, null);
    }

    public ResponseJSON(int status, Object obj, String msg) {
        this.status = status;
        this.obj = obj;
        this.msg = msg;
    }
}

File: gateway/src/main/kotlin/com/nurupo/movie/gateway/entity/ResponseJSON.kt
package com.nurupo.movie.gateway.entity

import com.fasterxml.jackson.annotation.JsonProperty

data class ResponseJSON(
        @JsonProperty("status")
        var status: Int = 0,

        @JsonProperty("obj")
        var obj: Any? = null,

        @JsonProperty("msg")
        var msg: String? = null
)